code/game/objects/items/devices/radio/beacon.dm

```diff
@@ -71,6 +71,19 @@
 		user.drop_item()
 		qdel(src)
 
+/obj/item/radio/beacon/syndicate/power_sink
+	name = "suspicious beacon"
+	desc = "A label on it reads: <i>Warning: Activating this device will send a power sink to your location</i>."
+	var/item_to_teleport = /obj/item/powersink
+
+/obj/item/radio/beacon/syndicate/power_sink/attack_self(mob/user)
```

Comment:
Since the change is made.
I think it would be better to have a spawning beacon type which does this code. But spawns a given object. Reduces the copypasta as well since now this same proc is made 3 times. 
By `/obj/item/radio/beacon/syndicate/attack_self`, `/obj/item/radio/beacon/syndicate/power_sink/attack_self` and `/obj/item/radio/beacon/syndicate/bomb/attack_self`
It would probably be better to make `/obj/item/radio/beacon/syndicate` an abstract generic class from which the others inherit.

Reply:
That can be done in another PR